### DISCUSSION TITLE
Using ts-standard to validate TypeScript

### DISCUSSION THREAD

**Discussion by dextermb** [react-hook-form/react-hook-form#9325] [Q&A] [answered]
Hey all 👋🏻 

I'm reasonably new to picking up TypeScript, and I'd love to get hands on and use it with react-hook-form. When writing in JavaScript I use the [`standard`](https://standardjs.com/) package for linting- there is a similar library for TypeScript called [`ts-standard`](https://github.com/standard/ts-standard).

---

But I've noticed it seems to flag an issue with the `handleSubmit` function:

<img width="944" alt="image" src="https://user-images.githubusercontent.com/6137789/199960162-d591538d-43dc-4919-9111-4d1efbb93961.png">

> Promise-returning function provided to attribute where a void return was expected. (@typescript-eslint/no-misused-promises)

What would be the best method to resolve this without using `// @ts-ignore` or `// @ts-expect-error`? I've currently got hinting to work by doing:

```ts
const onSubmit = function (event: BaseSyntheticEvent): void {
  handleSubmit(add)(event).finally(() => null)
}
```

_Note: The `.finally` is required (or `.then`/`.catch`) due to `ts-standard`_

---

Any good suggestions would be much appreciated.

**Comment by Moshyfawn** [comment]
Related #8622

**Reply by dextermb** [reply]
Thanks for sharing, still doesn't seem like a nice solution. Ideally I wouldn't have to wrap `form.handleSubmit` in anything.

**Comment by Moshyfawn** [comment, marked as answer]
Alright, I've done a little bit more research.

Check out this typescript-eslint/typescript-eslint#4623 PR that adds granular options within `checksVoidReturns` for [the `no-misused-promises` TS ESLint rule](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-misused-promises.md); the `attributes` options was added specifically for use-cases like the one you're experiencing. This is the official TS ESLint solution and we recommend you update your `.eslintrc` config file with it:

```json
"rules": {
  "@typescript-eslint/no-misused-promises": [2, {
    "checksVoidReturn": {
      "attributes": false
    }
  }]
},
```
